### ISSUE TITLE
What am I possibly doing wrong?

### ISSUE THREAD

**Issue by GeneralAwareness** [Linaqruf/kohya-trainer#50]
/content/kohya-trainer
+-------------------------------+-------------------------------------------------------------------------+
| Hyperparameter                | Value                                                                   |
+-------------------------------+-------------------------------------------------------------------------+
| v2                            | True                                                                    |
| v_parameterization            | False                                                                   |
| network_dim                   | 128                                                                     |
| network_alpha                 | 128                                                                     |
| network_module                | networks.lora                                                           |
| network_weights               | False                                                                   |
| network_train_on              | both                                                                    |
| unet_lr                       | 0.0001                                                                  |
| text_encoder_lr               | 5e-05                                                                   |
| no_metadata                   | True                                                                    |
| training_comment              | False                                                                   |
| lr_scheduler                  | constant                                                                |
| lr_scheduler_num_cycles       | 1                                                                       |
| lr_scheduler_power            | 1                                                                       |
| pretrained_model_name_or_path | /content/drive/MyDrive/pre_trained_model/stable-diffusion-2-1-768v.ckpt |
| vae                           | False                                                                   |
| caption_extension             | .txt                                                                    |
| train_folder_directory        | /content/drive/MyDrive/training_samples                                 |
| reg_folder_directory          | /content/drive/MyDrive/stable-diffusion-Regularization-Images/sd2.1/man |
| output_dir                    | /content/drive/MyDrive/dreambooth/output                                |
| prior_loss_weight             | 1.0                                                                     |
| resume_path                   | False                                                                   |
| project_name                  | Cry1n                                                                   |
| mixed_precision               | fp16                                                                    |
| save_precision                | fp16                                                                    |
| save_n_epochs_type            | save_n_epoch_ratio                                                      |
| save_n_epochs_type_value      | 5                                                                       |
| save_model_as                 | safetensors                                                             |
| resolution                    | 512                                                                     |
| train_batch_size              | 6                                                                       |
| max_token_length              | 225                                                                     |
| use_8bit_adam                 | True                                                                    |
| num_epochs                    | 5                                                                       |
| seed                          | 0                                                                       |
| gradient_checkpointing        | True                                                                    |
| gradient_accumulation_steps   | 1                                                                       |
| clip_skip                     | 2                                                                       |
| logging_dir                   | /content/dreambooth/logs                                                |
| log_prefix                    | Cry1n                                                                   |
| additional_argument           | --shuffle_caption --xformers                                            |
+-------------------------------+-------------------------------------------------------------------------+
2023-02-03 06:03:46.440222: E tensorflow/stream_executor/cuda/cuda_blas.cc:2981] Unable to register cuBLAS factory: Attempting to register factory for plugin cuBLAS when one has already been registered
2023-02-03 06:03:47.102590: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libnvinfer.so.7'; dlerror: libnvinfer.so.7: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: /usr/lib64-nvidia
2023-02-03 06:03:47.102695: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libnvinfer_plugin.so.7'; dlerror: libnvinfer_plugin.so.7: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: /usr/lib64-nvidia
2023-02-03 06:03:47.102714: W tensorflow/compiler/tf2tensorrt/utils/py_utils.cc:38] TF-TRT Warning: Cannot dlopen some TensorRT libraries. If you would like to use Nvidia GPU with TensorRT, please make sure the missing libraries mentioned above are installed properly.
2023-02-03 06:03:50.152932: E tensorflow/stream_executor/cuda/cuda_blas.cc:2981] Unable to register cuBLAS factory: Attempting to register factory for plugin cuBLAS when one has already been registered
2023-02-03 06:03:50.785035: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libnvinfer.so.7'; dlerror: libnvinfer.so.7: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: /usr/lib64-nvidia
2023-02-03 06:03:50.785144: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libnvinfer_plugin.so.7'; dlerror: libnvinfer_plugin.so.7: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: /usr/lib64-nvidia
2023-02-03 06:03:50.785162: W tensorflow/compiler/tf2tensorrt/utils/py_utils.cc:38] TF-TRT Warning: Cannot dlopen some TensorRT libraries. If you would like to use Nvidia GPU with TensorRT, please make sure the missing libraries mentioned above are installed properly.
prepare tokenizer
update token length: 225
Use DreamBooth method.
prepare train images.
0 train images with repeating.
prepare reg images.
found directory man contains 0 image files
0 reg images.
no regularization images / 正則化画像が見つかりませんでした
loading image sizes.
0it [00:00, ?it/s]
prepare dataset
No data found. Please verify arguments / 画像がありません。引数指定を確認してください

I get all the way to the training cell and it suddenly can't find my reg, nor my training, images while the other cells had no errors.  I tried the finetune one as well but it gave a weird error.

What I want is to be able to use my hand input captions (.txt) for my 768x768 images with SD 2.1.  Which would be the best notebook for that #1, and #2 what about the error above?

edit:  Is all that GPU stuff something that needs to be fixed and might be the cause?

**Comment by Linaqruf** [comment]
i assume it can read train data and reg data path but not the images inside, can you show me where you put your images?

**Comment by GeneralAwareness** [comment]
I put them on my gdrive that is mounted.

![image](https://user-images.githubusercontent.com/124339252/216534490-dd21d501-25b0-4997-8bfd-f9509ec225ac.png)

That one is the training and this one is the reg

![image](https://user-images.githubusercontent.com/124339252/216534711-03d7dfce-2378-416c-b3b6-0cc71fcd76c2.png)

**Comment by Linaqruf** [comment]
I'm sorry, but Kohya's dreambooth is different from another dreambooth script, you need to make file structure like this

```
|- reg_data
    |- <reg_repeats>_<class>

|- train_data
    |- <train_repeat>_<train_token> <class>
```
for example
```
|- reg_data
    |- 1_girl

|- train_data
    |- 10_skselsa girl
```

`skselsa` is the token word and `girl` is the class. In this example the regularization `girl` class images contained in the folder will be repeated only 1 time and the `skselsa` images will be repeated 10 times:

You can just rename your folder or put your images inside the created folder. Then set parent directory path.
![image](https://user-images.githubusercontent.com/50163983/216539841-5e56a8e4-bc49-4b5e-85bd-aa188de2e6f9.png)

**Comment by GeneralAwareness** [comment]
I am not understanding because I like to use this structure due to Shiv's and I share the data for across all kinds of training PLUS it makes it easier for me to just add my data.

Isn't there any way to allow it to just go to where I tell it?  Never seen it hard set like this before so seems very confining and strange too.

**Comment by GeneralAwareness** [comment]
Like this?

![image](https://user-images.githubusercontent.com/124339252/216543483-f9bae4a7-d405-45e7-b7f4-69142359ca0d.png)

**Comment by Linaqruf** [comment]
like this
![image](https://user-images.githubusercontent.com/50163983/216544656-862ceeee-35e3-4f77-80d4-2541cc13ddaf.png)
i'll try to figure out how to make this easier to use

**Comment by Linaqruf** [comment]
1 = how many repeats for your reg images
10 = how many repeats for your train images
skstoken = your initial prompt
man = class, just an identifier if you want to train on multiple concept

**Comment by GeneralAwareness** [comment]
I only use 1 repeat since that slows it way down and most I know have it set to 1 for train images and reg images I don't know.  What I do know is that I am having a very hard time grasping this even after looking at her help (after google translated) and her zip file.

**Comment by Linaqruf** [comment]
yeah you can do that, use repeat only if you train on small dataset

**Comment by Linaqruf** [comment]
it useful for multiple concept training like 
10_sksarcane style
20_sksdisney style
5_sksmcu style
24_sksharu character

but yea, this is too complicated for casual training

**Comment by GeneralAwareness** [comment]
My data is normally 16-30 images.  For finetuning that is micro.

**Comment by GeneralAwareness** [comment]
Seems for simple one concept stuff a repeat of 1 for class/reg and training is all that I need I was told.  What I need to know from you is this part - "yeah, you can just set it to 1 and set a higher interval for saving safetensor files".  How do I set a higher interval?

**Comment by GeneralAwareness** [comment]
number of images (including repeats) / 各bucketの画像枚数（繰り返し回数を含む）
bucket 0: resolution (256, 832), count: 0
bucket 1: resolution (256, 896), count: 0
bucket 2: resolution (256, 960), count: 0
bucket 3: resolution (256, 1024), count: 0
bucket 4: resolution (320, 704), count: 0
bucket 5: resolution (320, 768), count: 0
bucket 6: resolution (384, 640), count: 0
bucket 7: resolution (448, 576), count: 0
bucket 8: resolution (512, 512), count: 32
bucket 9: resolution (576, 448), count: 0
bucket 10: resolution (640, 384), count: 0
bucket 11: resolution (704, 320), count: 0
bucket 12: resolution (768, 320), count: 0
bucket 13: resolution (832, 256), count: 0
bucket 14: resolution (896, 256), count: 0
bucket 15: resolution (960, 256), count: 0
bucket 16: resolution (1024, 256), count: 0
mean ar error (without repeats): 0.0
prepare accelerator
Using accelerator 0.15.0 or above.
load StableDiffusion checkpoint
loading u-net: <All keys matched successfully>
loading vae: <All keys matched successfully>
loading text encoder: <All keys matched successfully>
Replace CrossAttention.forward to use xformers
caching latents.
100% 32/32 [00:19<00:00,  1.61it/s]
import network module: networks.lora
create LoRA for Text Encoder: 138 modules.
create LoRA for U-Net: 192 modules.
enable LoRA for text encoder
enable LoRA for U-Net
prepare optimizer, data loader etc.

===================================BUG REPORT===================================
Welcome to bitsandbytes. For bug reports, please submit your error trace to: https://github.com/TimDettmers/bitsandbytes/issues
For effortless bug reporting copy-paste your error into this form: https://docs.google.com/forms/d/e/1FAIpQLScPB8emS3Thkp66nvqwmjTEgxp8Y9ufuWTzFyr9kJ5AoI47dQ/viewform?usp=sf_link
================================================================================
CUDA_SETUP: WARNING! libcudart.so not found in any environmental path. Searching /usr/local/cuda/lib64...
CUDA SETUP: CUDA runtime path found: /usr/local/cuda/lib64/libcudart.so
CUDA SETUP: Highest compute capability among GPUs detected: 7.5
CUDA SETUP: Detected CUDA version 112
CUDA SETUP: Loading binary /usr/local/lib/python3.8/dist-packages/bitsandbytes/libbitsandbytes_cuda112.so...
use 8-bit Adam optimizer
override steps. steps for 5 epochs is / 指定エポックまでのステップ数: 30
running training / 学習開始
  num train images * repeats / 学習画像の数×繰り返し回数: 16
  num reg images / 正則化画像の数: 81
  num batches per epoch / 1epochのバッチ数: 6
  num epochs / epoch数: 5
  batch size per device / バッチサイズ: 6
  total train batch size (with parallel & distributed & accumulation) / 総バッチサイズ（並列学習、勾配合計含む）: 6
  gradient accumulation steps / 勾配を合計するステップ数 = 1
  total optimization steps / 学習ステップ数: 30
steps:   0% 0/30 [00:00<?, ?it/s]epoch 1/5

**Comment by Linaqruf** [comment]
nice
